### ISSUE TITLE
HackStudio: Deleting a file crashes the application

### ISSUE THREAD

**Issue by mrkct** [SerenityOS/serenity#9602]
1. Open a project
2. Right click on a file in the tree view
3. Delete the file
4. The program crashes with this stacktrace

```
CrashDaemon(21:21): --- Backtrace for thread #0 (TID 32) ---
CrashDaemon(21:21): 0x149f2cbd: [libgui.so] GUI::FileSystemModel::Node::index(int) const +0x2d (FileSystemModel.cpp:31)
CrashDaemon(21:21): 0x149f2dee: [libgui.so] GUI::FileSystemModel::parent_index(GUI::ModelIndex const&) const +0x6e (FileSystemModel.cpp:475)
CrashDaemon(21:21): 0x1497108e: [libgui.so] GUI::AbstractView::model_did_update(unsigned int) +0x58e (Model.h:85)
CrashDaemon(21:21): 0x1496d3f8: [libgui.so] GUI::AbstractTableView::model_did_update(unsigned int) +0x28 (AbstractTableView.cpp:342)
CrashDaemon(21:21): 0x14a99785: [libgui.so] GUI::TreeView::model_did_update(unsigned int) +0xb5 (TreeView.cpp:403)
CrashDaemon(21:21): 0x14a32ef9: [libgui.so] GUI::Model::for_each_client(AK::Function<void (GUI::ModelClient&)>) +0x79 (Function.h:91)
CrashDaemon(21:21): 0x14a33042: [libgui.so] GUI::Model::did_update(unsigned int) [clone .localalias] +0x62 (Model.cpp:54)
CrashDaemon(21:21): 0x149f606b: [libgui.so] GUI::FileSystemModel::handle_file_event(Core::FileWatcherEvent const&) [clone .part.0] +0x5b (FileSystemModel.cpp:435)
CrashDaemon(21:21): 0x14d95447: [libcore.so] AK::Function<void ()>::CallableWrapper<Core::FileWatcher::FileWatcher(int, AK::NonnullRefPtr<Core::Notifier>)::{lambda()#1}>::call() +0xf7 (Function.h:91)
CrashDaemon(21:21): 0x14da5107: [libcore.so] Core::Notifier::event(Core::Event&) [clone .localalias] +0xa7 (Function.h:91)
CrashDaemon(21:21): 0x14da66f2: [libcore.so] Core::Object::dispatch_event(Core::Event&, Core::Object*) +0xa2 (Object.cpp:213)
CrashDaemon(21:21): 0x14d8d12f: [libcore.so] Core::EventLoop::pump(Core::EventLoop::WaitMode) +0x3bf (EventLoop.cpp:395)
CrashDaemon(21:21): 0x14d8d8fd: [libcore.so] Core::EventLoop::exec() +0x5d (EventLoop.cpp:359)
CrashDaemon(21:21): 0x1497ae48: [libgui.so] GUI::Application::exec() +0x28 (Application.cpp:111)
CrashDaemon(21:21): 0x16d4d203: [/bin/HackStudio] main +0x753 (main.cpp:81)
CrashDaemon(21:21): 0x16d4d7c5: [/bin/HackStudio] _entry +0x65 (crt0.cpp:45)

```

**Comment by krkk** [comment]
I’ve made a possible fix in #9416, but it seems the issue is more related to the `GUI::FileSystemModel`.

**Comment by guerinoni** [comment]
This is something related with the current selected item in treeview! Also It crash when delete folder from file manager
